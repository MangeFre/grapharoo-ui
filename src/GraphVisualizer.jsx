import React, { Component } from 'react';
import './GraphVisualizer.css';
import URLNode from './URLNode';

// This can be used for CSS - Changing CSS
// should not cause rer-renders (I think)
export default class GraphVisualizer extends Component {
	constructor(props) {
		super();
		const { origin } = props;
		this.state = {
			nodes: [origin],
		};
		this.addNode = this.addNode.bind(this);
	}

	addNode(node) {
		const { nodes } = this.state;
		const newNodes = [...nodes];
		newNodes.push(node);
		this.setState({
			nodes: newNodes,
		});
	}

	render() {
		const { nodes } = this.state;
		const allNodes = nodes.map((nodeLink) => {
			return (
				<URLNode
					url={nodeLink}
					key={nodeLink}
					onFinishLoad={(node) => this.addNode(node)}
				/>
			);
		});
		return (
			<div>
				{allNodes}
				<style jsx>{`
					div {
						display: flex;
						flex-direction: column;
						width: 100%;
						height: 100%;
						background: #ff9999;
					}

					div:last-child {
						margin-bottom: 1rem;
					}
				`}</style>
			</div>
		);
	}
}
