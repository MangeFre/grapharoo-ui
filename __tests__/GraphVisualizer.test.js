import React from 'react';
import renderer from 'react-test-renderer';
import GraphVisualizer from '../src/GraphVisualizer';

const testLink = 'https://www.reddit.com/r/aww/comments/hd6xtp/comment/fvk5vao';

test('Create snapshot for future comparison', () => {
    const component = renderer.create(<GraphVisualizer origin={testLink}/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})