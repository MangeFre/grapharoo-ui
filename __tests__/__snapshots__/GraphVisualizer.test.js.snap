// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create snapshot for future comparison 1`] = `
<div
  className="jsx-2323080773"
>
  
</div>
`;
